autoload/dbext_dbi.vim

```diff
@@ -1302,7 +1428,12 @@ sub db_format_results
             # a way to check the maximum length of an array without checking
             # every entry which we are already doing here.
             foreach my $col ( @{$row} ) {
-                $temp_length = length((defined($col)?$col:""));
+                #$temp_length = length((defined($col)?$col:""));
+                # For some reason the above can sometimes return the wrong length.
+                # The following two lines fix that problem
+                my $xstr = $col;
+                $temp_length = length((defined($col)?$xstr:""));
+
```

Comment:
Great, thx

Reply:
Hm but this not comes from me. I took that other from v26, downloaded in 2021 from vim scripts site.
Looking on that precisely I see no real reason for this change. As far as I can tell lines 1434+1435 doing exactly the  same as line 1431 in the past?

(Well, single possible reason for difference which I can imagine is that encoding of the string is changed internally due to assignment but not the byte content of string - and it would be relevant only for some non-ascii bytes... But that's just a theory)